webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -2001,8 +2006,9 @@ trait PromiseGeneration_Hooks extends PromiseGenerationService with PromiseGener
     val (policyServers, simpleNodes) = {
       val (a, b) = updatedNodeConfigsInfo.values.toSeq.partition(_.isPolicyServer)
       (
-        // policy servers are sorted by their promiximity with root, root first
-        a.sortBy(x => NodePriority(x)).map(_.id.value), // simple nodes are sorted alphanum
+        // policy servers are sorted by their proximity with root, root first
+        a.sortBy(x => NodePriority(x.id, x.isPolicyServer, x.policyServerId))
```

Comment:
we're really just using these three attributes for priority, which are available on both `NodeInfo` and `CoreNodeFact`. It eases migration to be more specific here.